### DISCUSSION TITLE
some questions about AP APs APm

### DISCUSSION THREAD

**Discussion by Ashinhuang** [facebookresearch/detectron2#3067] [Q&A]
Dear sir,

![image](https://user-images.githubusercontent.com/33593302/118903403-4f0e7c80-b94a-11eb-95e0-0b8ce3cc886c.png)

I know that "AP50" is threshold value 0.5,   "AP75" is threshold value 0.75
but what is the "AP"  "APs"  "APm"  "APl"?
and "APs" appear Nan

**Comment by rvdinter** [comment]
Average Precision (AP) is measured in COCO-style. COCO provides more information about the metrics [here](https://cocodataset.org/#detection-eval). In short, AP stands for average precision measured over several thresholds, by default from 0.5 to 0.95, with a step size of 0.05. APs/m/l are about results regarding small, medium, and large instances, with an area of 32^2 and 96^2 as thresholds making a distinction between the object sizes. This is provided as some datasets, such as COCO have an imbalance in object sizes. For example, your results show good performance for medium and large instances, but no small objects have been detected (area <32^2).

**Comment by life97** [comment]
> Dear sir,
> 
> ![image](https://user-images.githubusercontent.com/33593302/118903403-4f0e7c80-b94a-11eb-95e0-0b8ce3cc886c.png)
> 
> I know that "AP50" is threshold value 0.5, "AP75" is threshold value 0.75
> but what is the "AP" "APs" "APm" "APl"?
> and "APs" appear Nan

Hello! How do I get the AP value for each predefined category of the object detection model when the IoU threshold is 0.5?
